### DISCUSSION TITLE
RK3588 Mainline Status

### DISCUSSION THREAD

**Discussion by kwankiu** [kwankiu/archlinux-installer-rock5#42] [General]
Available kernel for Arch Linux:
1. ~~`linux-aarch64-rk3588-collabora-git`~~
(No longer maintained, but you may still build it yourself)
~~- Collabora's RK3588 Hardware Enablement Kernel (pretty much mainline but with commit that is not merged to mainline)~~

2. `linux-aarch64-rockchip-armbian-git` (Currently the most usable option)
- Armbian's Linux Mainline (stable) kernel for aarch64 with armbian's patches for rockchip
- status: Pretty much usable, has GPU support, display output, with support for wifi firmware that is mainlined
- notes: contains patches from collabora, as well as support for various SBC like orange pi 5, 5b, khadas edge2, radxa rock 5a and 5b, and nanopi.

3. `linux-aarch64-7ji`
- 7Ji's Linux Mainline kernel for aarch64 with patches for rk3588 and amlogic
- status: unknown

4. `linux-aarch64`
- Arch Linux ARM's official Linux Mainline kernel for aarch64
- Usually 1-2 weeks behind the latest version listed on kernel.org
- status: no display output yet and contains dts of some popular rk3588 SBCs

**Comment by DarkevilPT** [comment]
My fan was working. I have just checked. So we have HDMI and FAN on Edge 2. What we dont have yet is Wifi and BT!

**Comment by kwankiu** [comment]
Comparing 1. `linux-aarch64-rk3588-collabora-git` and 2. `linux-aarch64-rockchip-armbian-git`.

collabora's is newer and contains commit that is not yet merged to mainline.

armbian's uses the stable mainline kernel with patches selected from collabora and added fixes and sbc support

if you want to try newer stuff, go for collabora, but it pretty much only supported the rock 5b, i have added armbian's rock 5a and khadas edge2 patch to this kernel so that it should works on rock 5a and 5b as well as khadas edge2, but there is no guarantee on functionality and collabora's branch is always updating, it can be broken at anytime.

therefore, if you want to know about the current status of mainline kernel, the armbian kernel is recommended.

**Reply by ageevp** [reply]
I've install 
https://github.com/hbiyik/agrrepo/releases/download/alarm-aarch64/linux-aarch64-rk3588-collabora-git-6.15.r1353554.33755850-1-aarch64.pkg.tar.xz
onto Orange pi 5 plus.
But there are issues
https://github.com/hbiyik/agrrepo/issues/12.
How to build
linux-aarch64-rk3588-collabora-git ?
Why Manjaro has no TROUBLES with mainline kernels?

**Comment by Petr1946** [comment]
Standard Non-UEFI sd card 
kernel 6.18.1-1 
works like a charm!!!!
Same UEFI edition
No any sound!!!!

**Reply by Petr1946** [reply]
solved
